core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -298,10 +298,10 @@ private[spark] class TaskSchedulerImpl(
       if (!executorIdToHost.contains(o.executorId)) {
```

Comment:
ok, sounds good.  I'm pretty sure there are a number of places where things wouldn't work if we did reuse executor Ids, but I agree its better to keep things consistent inside this class at least.